syllabus/examples/experimental/sb3_procgen_plr.py

```diff
@@ -1,38 +1,204 @@
+
+    def reset(self) -> np.ndarray:
+        obs = self.venv.reset()
+        assert isinstance(obs, dict)
+        return obs[self.key]
+
+    def step_wait(self) -> VecEnvStepReturn:
+        obs, reward, done, infos = self.venv.step_wait()
+        assert isinstance(obs, dict)
+        for info in infos:
+            if "terminal_observation" in info:
+                info["terminal_observation"] = info["terminal_observation"][self.key]
+        return obs[self.key], reward, done, infos
+
+
+def make_env(env_id, seed, curriculum=None, start_level=0, num_levels=1):
     def thunk():
-        env = gym.make("procgen-bigfish-v0", distribution_mode="easy", start_level=start_level, num_levels=num_levels)
-        env = ProcgenTaskWrapper(env)
-        env = MultiProcessingSyncWrapper(
-            env,
-            task_queue,
-            update_queue,
-            update_on_step=False,
-            task_space=env.task_space,
-        )
+        env = openai_gym.make(f"procgen-{env_id}-v0", distribution_mode="easy", start_level=start_level, num_levels=num_levels)
+        env = GymV21CompatibilityV0(env=env)
+        components = curriculum.get_components()
+        if curriculum is not None:
+            env = ProcgenTaskWrapper(env, env_id, seed=seed)
+            env = MultiProcessingSyncWrapper(
+                env=env,
+                components=components,
+                update_on_step=False,
+                task_space=env.task_space,
+            )
         return env
     return thunk
 
+def level_replay_evaluate_sb3(env_name, model, num_episodes, num_levels=0):
+
+    eval_envs = ProcgenEnv(
+    num_envs=args.num_eval_episodes, env_name=env_name, num_levels=num_levels, start_level=0, distribution_mode="easy", paint_vel_info=False
+    )
+
+    eval_envs = VecExtractDictObs(eval_envs, "rgb") 
+    eval_envs = wrap_vecenv(eval_envs)
+
+    eval_obs = eval_envs.reset()
+    eval_episode_rewards = [-1] * num_episodes
+
+    while -1 in eval_episode_rewards:
+        eval_action, _states = model.predict(eval_obs, deterministic=False) 
+
+        eval_obs, rewards, dones, infos = eval_envs.step(eval_action)
+        for i, info in enumerate(infos):
+            if 'episode' in info.keys() and eval_episode_rewards[i] == -1:
+                eval_episode_rewards[i] = info['episode']['r']
+
+    mean_returns = np.mean(eval_episode_rewards)
+    stddev_returns = np.std(eval_episode_rewards)
+    env_min, env_max = PROCGEN_RETURN_BOUNDS[args.env_id]
+    normalized_mean_returns = (mean_returns - env_min) / (env_max - env_min)
+    return mean_returns, stddev_returns, normalized_mean_returns
 
 def wrap_vecenv(vecenv):
-    vecenv.is_vector_env = True
     vecenv = VecMonitor(venv=vecenv, filename=None)
-    vecenv = VecNormalize(venv=vecenv, norm_obs=False, norm_reward=True)
+    vecenv = VecNormalize(venv=vecenv, norm_obs=False, norm_reward=True, training=False)
     return vecenv
```

Comment:
This function is used for both eval and training, we should probably pass training as an argument to wrap_vecenv, so that training=True for the training envs right?